src/object.rs

```diff
@@ -1109,92 +1016,43 @@ macro_rules! glib_object_wrapper {
     };
 
     // If there is no parent class, i.e. only glib::Object
-    (@munch_first_impl $name:ident, $rust_class_name:ident, ) => {
+    (@munch_first_impl $name:ident, ) => {
         $crate::glib_object_wrapper!(@munch_impls $name, );
-
-        impl ::std::ops::Deref for $rust_class_name {
-            type Target = <$crate::object::Object as $crate::object::ObjectType>::RustClassType;
-
-            fn deref(&self) -> &Self::Target {
-                $crate::object::IsClassFor::upcast_ref(self)
-            }
-        }
-
-        impl ::std::ops::DerefMut for $rust_class_name {
-            fn deref_mut(&mut self) -> &mut Self::Target {
-                $crate::object::IsClassFor::upcast_ref_mut(self)
-            }
+        unsafe impl $crate::object::SuperClassIs for $name {
+            type Super = $crate::object::Object;
         }
     };
 
     // If there is only one parent class
-    (@munch_first_impl $name:ident, $rust_class_name:ident, $super_name:path) => {
+    (@munch_first_impl $name:ident, $super_name:path) => {
         $crate::glib_object_wrapper!(@munch_impls $name, $super_name);
-
-        impl ::std::ops::Deref for $rust_class_name {
-            type Target = <$super_name as $crate::object::ObjectType>::RustClassType;
-
-            fn deref(&self) -> &Self::Target {
-                $crate::object::IsClassFor::upcast_ref(self)
-            }
-        }
-
-        impl ::std::ops::DerefMut for $rust_class_name {
-            fn deref_mut(&mut self) -> &mut Self::Target {
-                $crate::object::IsClassFor::upcast_ref_mut(self)
-            }
+        unsafe impl $crate::object::SuperClassIs for $name {
```

Comment:
This is now always only implemented for the direct parent class, or for the whole hierarchy?

Reply:
Only for the direct parent class, which then allows the type to `deref` recursively up to any parent type. Which seems to be how it worked before.